mu4/notation/view/notationtoolbarmodel.cpp

```diff
@@ -145,30 +161,105 @@ void NotationToolBarModel::updateState()
             item.checked = false;
         }
 
-        auto is = notation->interaction()->inputState();
-        if (is->isNoteEnterMode()) {
-            item("note-input").checked = true;
+        updateInputState();
+    }
 
-            item("pad-note-4").checked = is->duration() == DurationType::V_QUARTER;
-            item("pad-note-8").checked = is->duration() == DurationType::V_EIGHTH;
-            item("pad-note-16").checked = is->duration() == DurationType::V_16TH;
-        } else {
-            for (ActionItem& item : m_items) {
-                item.checked = false;
-            }
+    emit dataChanged(index(0), index(rowCount() - 1));
+}
+
+void NotationToolBarModel::updateInputState()
+{
+    auto inputState = notation()->interaction()->inputState();
+    if (!inputState->isNoteEnterMode()) {
+        for (ActionItem& item : m_items) {
+            item.checked = false;
         }
+
+        return;
     }
 
-    INotationUndoStackPtr undoStack = notation ? notation->undoStack() : nullptr;
-    if (undoStack) {
-        item("undo").enabled = undoStack->canUndo();
-        item("redo").enabled = undoStack->canRedo();
+    item("note-input").checked = true;
+
+    static QMap<actions::ActionName, Pad> noteInputActionPads = {
+        { "note-longa", Pad::NOTE00 },
+        { "note-breve", Pad::NOTE0 },
+        { "pad-note-1", Pad::NOTE1 },
+        { "pad-note-2", Pad::NOTE2 },
+        { "pad-note-4", Pad::NOTE4 },
+        { "pad-note-8", Pad::NOTE8 },
+        { "pad-note-16", Pad::NOTE16 },
+        { "pad-note-32", Pad::NOTE32 },
+        { "pad-note-64", Pad::NOTE64 },
+        { "pad-note-128", Pad::NOTE128 },
+        { "pad-note-256", Pad::NOTE256 },
+        { "pad-note-512", Pad::NOTE512 },
+        { "pad-note-1024", Pad::NOTE1024 },
+        { "pad-dot", Pad::DOT },
+        { "pad-dotdot", Pad::DOTDOT },
+        { "pad-dot3", Pad::DOT3 },
+        { "pad-dot4", Pad::DOT4 },
+        { "pad-rest", Pad::REST }
+    };
```

Comment:
I believe that this map should not be here, and it seems that somewhere already exists